static/js/tools/scatter2/chart.tsx

```diff
@@ -0,0 +1,564 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Chart component for retrieving, transforming, and plotting data.
```

Comment:
This component might be trying to do too much - I think it'll be easier to manage if the data loading happened elsewhere (in another class perhaps), then this component can focus on the presentation layer.

Reply:
separated into `ChartLoader` and `Chart`